### ISSUE TITLE
SIP submission should not incorporate the SIP

### ISSUE THREAD

**Issue by dnadales** [input-output-hk/decentralized-software-updates#37]
Currently the SIP submission signal incorporates the commit, which is basically a hash of the SIP:

```haskell
data IdeationPayload hashAlgo
  = Submit (SIPCommit hashAlgo) (SIP hashAlgo)
  | Reveal (SIP hashAlgo)

-- ...

  transitionRules = [
    do
      TRC ( participants
          , st@Data.State { commitedSIPs, submittedSIPs, revealedSIPs }
          , sig
          ) <- judgmentContext
      case sig of
        Submit sipc sip -> do
```

The reason why this is done like this at the moment is that when generating a reveal for a given commit, we need to know the commit that matches that hash.

The problem with this is that these rules cannot be used as an actual implementation since we in practice we cannot know the SIP that corresponds to a given commit, and also we have the runtime overhead of keeping a map.

## Proposed solution

See [this commit](https://github.com/input-output-hk/cardano-ledger-specs/pull/963/files#diff-605e0e7cdb8f5126b900ad08a29d668fR46) for an example of how this can be solved.

If the solution above is adopted, then we should make `participants :: Bimap Core.VKey Core.SKey` a phantom variable as well.

I (@dnadales) also tried the approach in which we add extra state to the generators, but this gave rise to a very complex trace generation (and shrinking) framework.

**Comment by dnadales** [comment]
Moving this from #35 to #65. We plan on using data-automata for generation. So removing the SIP from the submission should be even easier!

**Comment by dnadales** [comment]
I'm also un-assigning this from me since fixing this will be much easier, and no advanced Haskell skills are needed :) Not that I have them anyways.

**Comment by dnadales** [comment]
Solved in #142
